build/images/Dockerfile.build.ubi

```diff
@@ -12,30 +12,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-ARG BUILD_TAG
-FROM registry.access.redhat.com/ubi8 as antrea-build
```

Comment:
I am wondering why this ubi8 is no longer needed? 

@xliuxu FYI, this change will conflict with your PR https://github.com/antrea-io/antrea/pull/5737

Reply:
Thanks for the FYI. I can rebase the PR later.
BTW I think this change is to revert https://github.com/antrea-io/antrea/pull/5723

Reply:
@luolanzone I think it's because it no longer relies on glibc shipped with ubi8.

Reply:
That's correct. This change just reverts an earlier commit that I had to be introduced because of a glibc mismatch (I preserved the individual "revert" commits in the history for this PR, so you can take a look). Now that we disable cgo, the Antrea binaries no longer have a dependency on libc, and if there is a mismatch in version between the libc included in the golang Docker image and the libc included in the ubi Docker image, it doesn't matter anymore. Therefore we can revert to the golang Docker image, to avoid having to install Go manually.